### PR TITLE
Add ability to specify alternative text for media attachments

### LINKED ISSUE #117: Ability to add alt text to images/media
Images must have text alternatives that describe the information or function represented by the images. This ensures that images can be used by people with various disabilities. The main technique to add alt text is via the `alt` attribute.

Twitter added the ability to add alt text to images earlier this year. Something in Mastodon would be great!

Some more information on alt text:
- [Short note on coding alt text](https://www.paciellogroup.com/blog/2015/09/short-note-on-coding-alt-text/)
- [HTML 5.1 §4.7.5.1. Requirements for providing text to act as an alternative for images](https://www.w3.org/TR/html51/semantics-embedded-content.html#alt-text)
- [Web Content Accessibility Guideline 1.1 – Text Alternatives](https://www.w3.org/WAI/WCAG20/quickref/#qr-text-equiv-all)
- [Web Accessibility Tutorials Images Concepts](https://www.w3.org/WAI/tutorials/images/)

### PR DESCRIPTION
**Pull request by Gargron**
Fix #117

- POST /api/v1/media accepts `description` straight away
- PUT /api/v1/media/:id to update `description` (only for unattached ones)
- Serialized as `name` of Document object in ActivityPub
- Uploads form adjusted for better performance and description input

image
image

### CODE REVIEW

File: spec/models/media_attachment_spec.rb [hunk 1]
```diff
@@ -50,4 +49,12 @@
       expect(media.file.meta["small"]["aspect"]).to eq 400.0/267
     end
   end
+
+  describe 'descriptions for remote attachments' do
+    it 'are cut off at 140 characters' do
```
Comment (beatrix-bitrot) [inline spec/models/media_attachment_spec.rb]: are -> is

Comment (Gargron) [reply spec/models/media_attachment_spec.rb]: descriptions for remote attachments -> are cut off at 140 characters

Comment (beatrix-bitrot) [reply spec/models/media_attachment_spec.rb]: yeah but 'it'...

**Review by beatrix-bitrot — APPROVED** [review]
approved with caveat that some tweaks will likely be required before release

**Comment by pwoolcoc** [comment]
I'm so so happy this finally got in! 🎆 🎉 

I was just curious, is there a reason for the 140-character limit?

**Comment by jennifermf** [comment]
Thank you for working on this!! This is really appreciated! And I'm sorry to be picky and never pleased, but is there any way we can bump up that 140-char limit? As a general rule, that's probably good for a real general description ("person holding a cat," "sleeping cat"), but for more detailed stuff, 500 char would be ideal and hopefully encourage people to use it... since many of us know the frustration of trying to cram a decent description of anything into 140 chars ;)

**Comment by Sylvhem** [comment]
Thank you so much for you work on that! It was a really needed feature.

I think the 140 characters limit looks kinda weird thought. 140 seems like a random number to me. Do you think it's possible to bump it to 150?

**Comment by MichielBijl** [comment]
The 140 char limit is really low. Twitter uses 420.

**Pull request merged**